ngraph/core/src/validation_util.cpp

```diff
@@ -1084,6 +1086,54 @@ namespace
         // Noting we can do
         return {MaxValue(data.m_value)};
     }
+
+    vector<MaxValue> exec_shape_of(Node* node, vector<MaxValue>& inputs)
+    {
+        const auto& inputPS = node->get_input_partial_shape(0);
+        std::vector<uint64_t> shapeDims;
+        for (size_t i = 0; i < inputPS.rank().get_length(); i++)
+        {
+            if (inputPS[i].is_static())
+            {
+                shapeDims.push_back(inputPS[i].get_length());
+            }
+            else
+            {
+                shapeDims.push_back(std::numeric_limits<uint64_t>::max());
+            }
+        }
+
+        return {MaxValue(shapeDims, 0)};
+    }
+
+    vector<MaxValue> exec_gather(Node* node, vector<MaxValue>& inputs)
```

Comment:
Why do we need these methods for gather and shape_of?

Reply:
This method is used to propagate the maximum value for K in TopK and for dynamic cases on VPU we are required to calculate this value, because it will give us the upper-bound of TopK's output. The networks are mentioned in the ticket has `ShapeOf` and `Gather` nodes on the K evaluation subgraph, so we need these methods to propagate K for these networks correctly.